metadata.rb

```diff
@@ -23,12 +23,15 @@
   supports os
 end
 
-depends          'apt' # We recommend '>= 2.1.0'. See CHANGELOG.md for details
-depends          'chef_handler', '>= 1.2' # We recommend '~> 1.3' with Chef < 12. See CHANGELOG.md for details
-depends          'windows' # We recommend '< 1.39.0' if running Chef >= 12.6. See README.md for details
-depends          'yum', '>= 3.0' # Use '~> 3.0' with Chef < 12
-
-suggests         'sudo' # ~FC052
+depends    'chef_handler'
+# In order to support Chef < 12.9, we need apt_repository which is available
+# in apt < 6.0.0. This will be removed for chef-datadog 4.0 while dropping the support
+# of Chef 12.
+depends    'apt', '< 6.0.0'
```

Comment:
@remeh the dependency constraints we define in `metadata.rb` should be as loose as possible, since they're constraints that are applied on the whole Chef environment of a given user.
If we set the `apt` constraint only to support Chef < 12.9, we should not set the constraint here but instead document this additional constraint in the readme for Chef 12.9 users. For example, this will allow Chef 14 users to still use the `datadog` cookbook with apt `6.0`.
Does that make sense? Happy to discuss this offline as well